### PR TITLE
During update, download updates first before removing old plugin files 

### PR DESCRIPTION
**Pull request by TrainDoctor**

### CODE REVIEW

File: frontend/src/toaster.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ class Toaster extends Logger {
       await sleep(2000);
     }
 
-    this.node = instance.return.return;
+    this.node = instance.sibling.child;
```
Comment (AAGaming00) [inline frontend/src/toaster.tsx:41]: incorrect merge, should be
```suggestion
    this.node = instance.return.return;
```

File: backend/browser.py [hunk 2]
```diff
@@ -97,7 +98,7 @@ def find_plugin_folder(self, name):
                     plugin = json.load(f)
 
                 if plugin['name'] == name:
-                    return path.join(self.plugin_path, folder)
+                    return String(path.join(self.plugin_path, folder))
```
[the remark this answers is not in the document]
Comment (TrainDoctor) [reply backend/browser.py:101]: honestly, no idea. I'll change it later.

**Comment by TrainDoctor** [comment]
Rebased it to the basic fixes that @AAGaming00 has done.

**Comment by AAGaming00** [comment]
Please do not rebase until that branch is ready.

**Comment by TrainDoctor** [comment]
> Please do not rebase until that branch is ready.

I'm aware, the rebase is for temporary testing to verify my changes for sanity. This will be reverted once testing is complete.

**Comment by TrainDoctor** [comment]
Alright with help from @Jaynator495 and @suchmememanyskill I've done some extensive testing and even fixed a bonus bug in the process.

**Review by Jaynator495 — APPROVED** [review]
Tested and works

**Review by suchmememanyskill — APPROVED** [review]

**Comment by TrainDoctor** [comment]
Further verified and tested with Whip on discord. Merging to main!

**Pull request merged**